xls/contrib/mlir/IR/xls_ops.td

```diff
@@ -1881,6 +1993,48 @@ def Xls_TraceOp : Xls_Op<"trace", [TensorArrayTypeFungible,
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// Other side-effecting operations
+//===----------------------------------------------------------------------===//
+
+def Xls_GateOp :
+  Xls_Op<"gate", [
+    ShapesAreConsistent<["data", "result"]>,
+    Elementwise,
+    Scalarizable,
+]> {
+  let summary = "Gates an arbitrarily-typed value based on a condition.";
```

Comment:
nit: remove period at the end of summary (it is used to autogenerate error messages).